libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -925,19 +925,34 @@ void AP_Baro::update(void)
     }
 #ifndef HAL_BUILD_AP_PERIPH
     const uint32_t now_ms = AP_HAL::millis();
-    if (now_ms - _field_elevation_last_ms >= 1000 && fabsf(_field_elevation_active-_field_elevation) > 1.0) {
-      if (!AP::arming().is_armed()) {
+    bool new_field_elev = false;
```

Comment:
Should hae an `update_field_elevation` method

Reply:
done